setup.py

```diff
@@ -48,14 +48,16 @@
     url='https://github.com/ethpm/py-ethpm',
     include_package_data=True,
     install_requires=[
-        'cytoolz==0.9.0',
-        'eth-keys==0.2.0b3',
-        'eth-tester==0.1.0b24',
-        'eth-utils==1.0.2',
-        'jsonschema==2.6.0',
-        'py-evm==0.2.0a11',
-        'rlp==0.6.0',
-        'web3==4.2.0',
+        "toolz>=0.9.0,<1.0.0;implementation_name=='pypy'",
+        "cytoolz>=0.9.0,<1.0.0;implementation_name=='cpython'",
+        'eth-keys>=0.2.0b3',
+        'eth-tester<0.1.0b25,>=0.1.0b24',
+        'eth-utils>=1.0.2',
+        'jsonschema>=2.6.0',
+        'py-evm==0.2.0a16',
+        'py-solc>=2.1.0',
+        'rlp<2.0.0,>=1.0.1',
+        'web3>=4.2.1',
```

Comment:
These should all be limited within major version ranges, like:

```py
        'web3>=4.2.1,<5',
```

The APIs might change at each major version change.